src/backend/distributed/operations/shard_split.c

```diff
@@ -375,13 +392,89 @@ SplitShard(SplitMode splitMode,
 							   SplitOperationName[splitOperation],
 							   SplitTargetName[splitOperation])));
 	}
-	else if (PlacementMovedUsingLogicalReplicationInTX)
+}
+
+
+/*
+ * ErrorIfMultipleNonblockingSplitInTheSameTransaction will error if we detect multiple
+ * nonblocking split operation in the same transaction.
+ */
+static void
+ErrorIfMultipleNonblockingSplitInTheSameTransaction(SplitOperation splitOperation)
+{
+	if (PlacementMovedUsingLogicalReplicationInTX)
 	{
 		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-						errmsg("multiple shard movements/splits via logical "
+						errmsg("multiple '%s %s' operations via logical "
 							   "replication in the same transaction is currently "
-							   "not supported")));
+							   "not supported",
+							   SplitOperationName[splitOperation],
+							   SplitTargetName[splitOperation]),
+						errhint("If you wish to execute multiple '%s %s' operations "
+								"in a single transaction set the shard_transfer_mode "
+								"to 'block_writes'.",
+								SplitOperationName[splitOperation],
+								SplitTargetName[splitOperation])));
 	}
+}
+
+
+/*
+ * GetWorkerNodesFromWorkerIds returns list of worker nodes given a list
+ * of worker ids. It will error if any node id is invalid.
+ */
+static List *
+GetWorkerNodesFromWorkerIds(List *nodeIdsForPlacementList)
+{
+	List *workersForPlacementList = NIL;
+	Datum nodeId;
+	foreach_int(nodeId, nodeIdsForPlacementList)
+	{
+		uint32 nodeIdValue = DatumGetUInt32(nodeId);
```

Comment:
not this PR's fault, but it's quite unexpected to me that we have Datums in here. 

We also don't really expect foreach_int to work on a list of datums. It just so happens that the Datum representation is the same as the int representation. We should fix that and generate a list of ints in IntegerArrayTypeToList